osm-fudge/bk_tree.py

```diff
@@ -0,0 +1,37 @@
+'''An implementation of bk-tree
+
+More: https://en.wikipedia.org/wiki/BK-tree
+'''
+
+import levenshtein
+
+class BKTree:
+    '''A bk-tree implementation and defined user methods
+       for insertion and querying the string
+    '''
+
+    def __init__(self, root):
+        self.tree = {root: {}}
```

Comment:
If you check the resources I sent you, you will see that BK-Trees can work with an arbitrary metric. The Levenshtein distance is only one example of such a metric. If you add an argument to your BK-Tree and pass in the metric from the outside, the BK-Tree is not bound to the concrete metric.